emukit/core/loop/loop_state.py

```diff
@@ -9,6 +9,23 @@
 from .user_function_result import UserFunctionResult
 
 
+def create_loop_state(x_init: np.ndarray, y_init: np.ndarray) -> None:
```

Comment:
When used in a loop with cost, we may want to save in the loop state the values of the cost at each x. For instance, if the cost model is accounting for the time of running an evaluation of the user objective. I think it makes sense having some LoopStateWithCost to handle that.

Reply:
I am also not happy about this right now. `LoopState` already has a property C which stores the cost of each function value. However, `create_loop_state()` only passes X / Y to the `LoopState`. Should we extent `create_loop_state(x_init , y_init)` to `create_loop_state(x_init , y_init, c_init=None)` ?

Reply:
That plan sounds good to me. Two things: this PR seems to undo a previous PR I made where I fixed the return type for this function. Can we call it `cost` rather than `c` everywhere please?

Reply:
A general comment to the name. We seem to call everything "cost" not matter if it is just a re-weighing of the acquisition point because for whatever reason we want to enforce certain points to be evaluated more often and other less. 

And second, a "cost" in the classic sense which influences the budget we already used to determine the stopping of the loop. So we kind of have three different scenarios:

1) a custom-weighing of candidate points, evaluation-cost same for all points
2) an evaluation-cost for each point which also determines the budget
3) a combination of 1) and 2)

Algebraically, 1) and 2) they are the same for the acquisition function, but 3) is not included. Also 1) and 2) will need to be handled differently by the outer loop. Thoughts?